go/types/struct.go

```diff
@@ -195,13 +210,45 @@ func EscapeStructField(input string) string {
 		buf.WriteString(hs)
 		return buf.String()
 	}
+	output := EscapeFields(input, encode)
+	verifyFieldName(output)
```

Comment:
Correct, I suppose if we were to break the encoding function, we'd catch something like that in test in general. Wasn't sure if really bad things happened if we introduced a bug in the encode that was missed in test, but that generated bad data or if we'd have enough checks upstream that we'd just fail to write the struct.